license_sh/analyze/analyze_shared.py

```diff
@@ -10,7 +10,8 @@
 from anytree import AnyNode, PreOrderIter
 
 from license_sh.analyze import lib
-from ..helpers import get_node_id
+
+NODE_ID_SEP = ":-:"
```

Comment:
We should get rid of this separator and support native separators instead `==` or `@` instead + support pure functions. Or if we want to do it in another PR, please leave the TODO and corresponding issue.

Reply:
It is just used in 2 functions that are decoding and encoding name and version. It is nowhere saved

Reply:
@S4n60w3n if that is the case, then I'd consider using some other data structure. e.g. Tuple or NamedTuple.

Reply:
Don't know what examply you mean but i have updated it a bit